File: v1/src/main/java/com/api/v1/doctors/services/DoctorManagementService.java
package com.api.v1.doctors.services;

import com.api.v1.people.requests.PersonUpdateDTO;
import org.springframework.http.ResponseEntity;

public interface DoctorManagementService {
    ResponseEntity<Void> terminate(String licenseNumber);
    ResponseEntity<Void> rehire(String licenseNumber);
    ResponseEntity<Void> update(String licenseNumber, PersonUpdateDTO personUpdateDTO);
}

File: v2/src/main/kotlin/com/api/v2/doctors/services/DoctorManagementService.kt
package com.api.v2.doctors.services

import com.api.v2.people.PersonUpdateDTO
import org.springframework.http.ResponseEntity

interface DoctorManagementService {
    fun terminate(licenseNumber: String): ResponseEntity<Unit>
    fun rehire(licenseNumber: String): ResponseEntity<Unit>
    fun update(licenseNumber: String, updateDTO: PersonUpdateDTO): ResponseEntity<Unit>
}